src/main/java/com/project/springbootproject/service/user/UserServiceImpl.java

```diff
@@ -20,23 +24,30 @@ public class UserServiceImpl implements UserService {
     private final UserMapper userMapper;
     private final PasswordEncoder passwordEncoder;
     private final RoleRepository roleRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
 
     @Override
+    @Transactional
     public UserDto register(UserRequestDto userRequestDto) {
         if (userRepository.findByEmail(userRequestDto.getEmail()).isPresent()) {
             throw new RegistrationException("Can not register user");
         }
 
-        User user = new User();
+        final User user = new User();
         user.setEmail(userRequestDto.getEmail());
         user.setPassword(passwordEncoder.encode(userRequestDto.getPassword()));
         user.setFirstName(userRequestDto.getFirstName());
         user.setLastName(userRequestDto.getLastName());
         user.setShippingAddress(userRequestDto.getShippingAddress());
-        Role userRole = roleRepository.findByRole(Role.RoleName.USER).orElseThrow(
+        final Role userRole = roleRepository.findByRole(Role.RoleName.USER).orElseThrow(
                 () -> new IllegalStateException("Role User not found"));
         user.setRoles(Collections.singleton(userRole));
-        User saveUser = userRepository.save(user);
+        final User saveUser = userRepository.save(user);
+        final ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCart.setCartItems(new HashSet<>());
+        shoppingCartRepository.save(shoppingCart);
         return userMapper.toUserDto(saveUser);
```

Comment:
method became too big, create a separate private method for dealing with shopping cart 
instead of setters, consider using mapstruct mapper

Reply:
The use of setters in this method was in Bogdan's training video